### DISCUSSION TITLE
Keep history of edits (SCD2)

### DISCUSSION THREAD

**Discussion by geoHeil** [directus/directus#18057] [Archive]
How can a collection containing user_updated_at and update_timestamp (the default system fields), not only track the current state but keep the old one (SCD2) to keep history?

**Comment by geoHeil** [comment]
https://github.com/directus/directus/discussions/2975 seems to be a bit similar

**Comment by br41nslug** [comment]
This is what revisions do when enabled https://docs.directus.io/getting-started/glossary.html#revisions

**Reply by geoHeil** [reply]
This is not exactly the same thing. Revisions are sort of a CDC stream to a central table.
What I mean/want is within the same table keeping the records with valid_from/valid_to and not jsonifying it all in one single mixed revisions table

**Comment by rijkvanzanten** [comment]
Question that I've been struggling with as part of https://github.com/directus/directus/discussions/2975! How would that work with things like unique fields or foreign key constraints within junction tables, and with that relations in general? If you make a change to a nested relational item as part of the same update, would you expect the updated row to exist in the related table as well?

**Reply by geoHeil** [reply]
great question. I think ideally that only the copy/SCD2 new row is emitted for each collection individually.
However, you must ensure that temporal join fields allow to join them back in a historically correct way
